### PR TITLE
Test `dartsass:install` command 

### PR DESCRIPTION
**Pull request by jonathanhefner**
This PR _looks_ large because this project currently has no testing infrastructure.  The vast majority of LOC in this PR are from `Gemfile` and `Gemfile.lock` files.  I have split the PR into two commits to further clarify.  The latter commit has the actual testing code, which amounts to ~70 lines of test code + ~70 lines of helper code.

---

This adds test coverage for the `dartsass:install` command.  The command is tested against a freshly generated Rails app using the version of Rails that is currently loaded.  Thus the installer can be tested with different versions of Rails in CI.

Infrastructure has been added to run tests against Rails 6.1, 7.0, 7.1, and `main` branch, all with both Sprockets and Propshaft.

### CODE REVIEW

**Comment by jonathanhefner** [comment]
I am seeing test failures when running the GitHub CI workflow in my own fork, but only with Ruby 3.0 and Rails >= 7.1.  I'm not sure what is happening, but it's something related to Bundler (the `bundle` command is the point of failure).  I will investigate further.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+
+      # This prevents a "Failed to build gem native extension" error when
+      # running `bundle install` inside a Rails app that is generated for
+      # testing. The error occurs only in CI, and only when using Ruby 3.0 and
+      # Rails >= 7.1.
+      - name: Install sass-embedded gem
+        if: ${{ matrix.ruby-version == '3.0' }}
+        run: gem install sass-embedded
```
Comment (jonathanhefner) [inline .github/workflows/ci.yml:48]: The underlying cause is unclear, but this is the resolution to [my previous comment](https://github.com/rails/dartsass-rails/pull/49#issuecomment-1837293293).

**Review by rud — COMMENTED** [review]
This is **magnificent**!

**Comment by dhh** [comment]
@jonathanhefner did you get all the variations to pass?

**Comment by jonathanhefner** [comment]
@dhh Yes: https://github.com/jonathanhefner/dartsass-rails/actions/runs/7089553301.

I did not merge because I do not have commit permissions for this repo. :smile:

**Pull request merged**